### PR TITLE
[Testing] Updated to latest RC.6 of Appium

### LINKED ISSUE #20624: [Testing] Update to latest RC.5 of Appium
### Description

We're using a number of obsolete APIs with the latest Appium
Let's get updated to RC.5 so we stop adding more work for ourselves long term and hopefully benefit from stability improvements with appium .

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Updated to latest RC.6 of Appium. Changes:
- Updated versions to use appium rc6.
- Replaced the usaged of TouchActions by ActionSequences.
- Changes to Reset the App.

### Issues Fixed

Fixes #20624

### CODE REVIEW

File: src/Controls/tests/UITests/Tests/Issues/Issue17610.cs [hunk 1]
```diff
@@ -31,17 +33,19 @@ public void RefreshIconDisappearsWhenUserCancelsRefreshByScrollingBackUp()
...
+			int toY = rect2.CenterY();
+
+			OpenQA.Selenium.Appium.Interactions.PointerInputDevice touchDevice = new OpenQA.Selenium.Appium.Interactions.PointerInputDevice(PointerKind.Touch);
+			var refreshSequence = new ActionSequence(touchDevice, 0);
+			refreshSequence.AddAction(touchDevice.CreatePointerMove(CoordinateOrigin.Viewport, fromX, fromY, TimeSpan.Zero));
+			refreshSequence.AddAction(touchDevice.CreatePointerDown(PointerButton.TouchContact));
+			refreshSequence.AddAction(touchDevice.CreatePointerMove(CoordinateOrigin.Viewport, toX, toY, TimeSpan.FromMilliseconds(250)));
+			refreshSequence.AddAction(touchDevice.CreatePointerMove(CoordinateOrigin.Viewport, fromX, fromY, TimeSpan.FromMilliseconds(250)));
+			refreshSequence.AddAction(touchDevice.CreatePointerUp(PointerButton.TouchContact));
+			androidApp.Driver.PerformActions([refreshSequence]);
```
Comment (jfversluis) [inline src/Controls/tests/UITests/Tests/Issues/Issue17610.cs:48]: Just out of curiosity, why is this better? Same question applies to all the other changes I guess 😄

Comment (jsuarezruiz) [reply src/Controls/tests/UITests/Tests/Issues/Issue17610.cs:48]: `TouchAction` is deprecated.

File: src/TestUtils/src/UITest.Appium/Actions/AppiumLifecycleActions.cs [hunk 2]
```diff
@@ -79,7 +82,9 @@ CommandResponse CloseApp(IDictionary<string, object> parameters)
 			if (_app?.Driver is null)
 				return CommandResponse.FailedEmptyResponse;
 
+#pragma warning disable CS0618 // Type or member is obsolete
 			_app.Driver.CloseApp();
```
[the remark this answers is not in the document]
Comment (jsuarezruiz) [reply src/TestUtils/src/UITest.Appium/Actions/AppiumLifecycleActions.cs]: Yes, the replacement is the `TerminateApp `method.

File: src/TestUtils/src/UITest.Appium/Actions/AppiumLifecycleActions.cs [hunk 3]
```diff
@@ -49,7 +49,9 @@ CommandResponse LaunchApp(IDictionary<string, object> parameters)
 			if (_app?.Driver is null)
 				return CommandResponse.FailedEmptyResponse;
 
+#pragma warning disable CS0618 // Type or member is obsolete
 			_app.Driver.LaunchApp();
```
[the remark this answers is not in the document]
Comment (jsuarezruiz) [reply src/TestUtils/src/UITest.Appium/Actions/AppiumLifecycleActions.cs]: Yes, ActivateApp.

**Comment by jsuarezruiz** [comment]
The build is failing passing the Android UI Test all the time in the Issue18896Test:
Issue18896Test-diff
Now, the first cell is selected (but we are not setting the SelectedItem property). It is not related to the changes in this PR.

**Comment by jfversluis** [comment]
Seems there are some conflicts now?

File: src/TestUtils/src/UITest.Appium/Actions/AppiumScrollActions.cs [hunk 4]
```diff
@@ -147,12 +149,13 @@ static void ScrollToLeft(AppiumDriver driver, AppiumElement element, ScrollStrat
...
-				.Press(startX, startY)
-				.Wait(strategy != ScrollStrategy.Programmatically ? swipeSpeed : 0)
-				.MoveTo(endX, endY)
-				.Release()
-				.Perform();
+			OpenQA.Selenium.Appium.Interactions.PointerInputDevice touchDevice = new OpenQA.Selenium.Appium.Interactions.PointerInputDevice(PointerKind.Touch);
+			var scrollSequence = new ActionSequence(touchDevice, 0);
+			scrollSequence.AddAction(touchDevice.CreatePointerMove(CoordinateOrigin.Viewport, startX, startY, TimeSpan.Zero));
+			scrollSequence.AddAction(touchDevice.CreatePointerDown(PointerButton.TouchContact));
+			scrollSequence.AddAction(touchDevice.CreatePointerMove(CoordinateOrigin.Viewport, endX, endY, TimeSpan.FromMicroseconds(strategy != ScrollStrategy.Programmatically ? swipeSpeed : 0)));
```
Comment (PureWeen) [inline src/TestUtils/src/UITest.Appium/Actions/AppiumScrollActions.cs:156]: I think these timespans are too fast

If I slow down the move to a second then it scrolls just fine.

**Comment by jsuarezruiz** [comment]
Updated from RC5 to RC6.

**Comment by jsuarezruiz** [comment]
Reviewed the ui tests related to scroll.
fix-scroll-appium-tests

File: src/TestUtils/src/UITest.Appium/Actions/AppiumScrollActions.cs (hunk 4 above)
Comment (jsuarezruiz) [reply src/TestUtils/src/UITest.Appium/Actions/AppiumScrollActions.cs]: Fixed.

File: src/Controls/tests/UITests/Tests/LabelUITests.cs [hunk 5]
```diff
@@ -32,6 +32,11 @@ public void SpanTapped()
 			Assert.Ignore("Click (x, y) pointer type mouse is not implemented.");
 		}
 
+		if (Device == TestDevice.Windows)
+		{
+			Assert.Ignore("GetRect not implemented on Windows.");
```
Comment (jsuarezruiz) [inline src/Controls/tests/UITests/Tests/LabelUITests.cs:37]:
```
[WinAppDriver] Command not implemented: GET: /session/3B433044-76E3-4017-BEAC-4911755824F9/element/42.6554534.4.405/rect
[WD Proxy] Got response with status 501: Command not implemented: GET: /session/3B433044-76E3-4017-BEAC-4911755824F9/element/42.6554534.4.405/rect
[W3C (22c5ea62)] Encountered internal error running command: Error: Could not proxy. Proxy error: Request failed with status code 501
```

Ignore Windows tests using the GetRect method https://w3c.github.io/webdriver/#get-element-rect
Getting an not implemented exception. Was working in previous versions.

File: src/TestUtils/src/UITest.Appium/Actions/AppiumLifecycleActions.cs [hunk 6]
```diff
@@ -93,24 +93,19 @@ CommandResponse ResetApp(IDictionary<string, object> parameters)
...
+			if (_app.GetTestDevice() == TestDevice.Windows)
 			{
-				_app.Driver.ResetApp();
-			}
-			else if (_app.GetTestDevice() == TestDevice.Windows)
-			{
-				CloseApp(parameters);
+				if (!string.IsNullOrEmpty(_app.Driver.CurrentWindowHandle))
+				{
+					_app.Driver.CloseApp();
```
Comment (jsuarezruiz) [inline src/TestUtils/src/UITest.Appium/Actions/AppiumLifecycleActions.cs:100]: Changes on Windows:
- TerminateApp -> NotImplementedException
- ActivateApp -> NotImplementedException

**Comment by jsuarezruiz** [comment]
Converted to Draft while reviewing Windows UITest and the lifecycle of the App Window.

File: src/Controls/tests/UITests/Tests/Issues/Issue19956.cs [hunk 7]
```diff
@@ -74,14 +83,40 @@ public void BottomInsetsSetCorrectly()
...
-        // scroll up once to trigger resetting content insets
-        actions.LongPress(null, 5, 300).MoveTo(null, 5, 450).Release().Perform();
-        actions.LongPress(null, 5, 400).MoveTo(null, 5, 100).Release().Perform();
-
-        actions.LongPress(null, 5, 400).MoveTo(null, 5, 100).Release().Perform();
-        actions.LongPress(null, 5, 400).MoveTo(null, 5, 100).Release().Perform();
+	{
+		OpenQA.Selenium.Appium.Interactions.PointerInputDevice touchDevice = new OpenQA.Selenium.Appium.Interactions.PointerInputDevice(PointerKind.Touch);
+		
+		var scrollSequence1 = new ActionSequence(touchDevice, 0);
```
Comment (PureWeen) [inline src/Controls/tests/UITests/Tests/Issues/Issue19956.cs:89]: Can you consolidate `LongPress` behind something re-usable? I think we can add an extension method to `Iapp` that we can reuse across all these tests yea?

**Review by PureWeen — CHANGES_REQUESTED** [review]

**Review by PureWeen — CHANGES_REQUESTED** [review]
Catalyst is still failing but WinUI looks good

**Review by PureWeen — APPROVED** [review]

**Comment by PureWeen** [comment]
Failing tests are unrelated. They are just part of the device tests these changes are only ui tests.

**Pull request merged**